src/pages/Game.tsx

```diff
@@ -146,15 +147,43 @@ const GamePage = ({ ws, currentUser }: GameProps): JSX.Element => {
    */
   const gameStartInit = (payload: unknown): void => {
     const data = payload as GameStartJSON;
-    const tileArray = data.tiles;
+    const { tiles: tileArray, selfPlayedTiles, currentIndex } = data;
+
+    const mjGameState = gameState as MahjongGameState;
+
+    // Increment wall counter
+    const wallCounter = mjGameState.getWallCounter();
+    const result = wallCounter.setCurrentIndex(currentIndex);
+    if (!result) console.error('Failed to set wall counter. Please verify game state');
+
+    // Set player hand
     const tiles: Tile[] = [];
     tileArray.forEach((tile: string) => {
       tiles.push(TileFactory.createTileFromStringDef(tile));
     });
-    const mjGameState = gameState as MahjongGameState;
-
-    const mjPlayer = mjGameState.getMjPlayer();
+    const mjPlayer = mjGameState.getMjPlayer() as MahjongPlayer;
     mjPlayer.setHand(tiles);
+    const mjPlayerHand = mjPlayer.getHand();
+
+    // Set player played tiles (bonus tiles) at game start
+    const mjPlayerBonusTilesStrDef = selfPlayedTiles.filter(
+      (playedTiles: SelfPlayedTile) => playedTiles.connectionId === mjPlayer.getConnectionId(),
+    )[0].playedTiles;
+    const mjPlayerBonusTiles = mjPlayerBonusTilesStrDef.map((strDef) => TileFactory.createTileFromStringDef(strDef));
+    mjPlayerHand.addPlayedTiles(mjPlayerBonusTiles);
+
+    // Set opponents played tiles s tiles) at game start
+    const mjOpponents = mjGameState
+      .getUsers()
+      .filter((user) => user.getConnectionId() !== mjPlayer.getConnectionId()) as MahjongOpponent[];
+    const opponentsBonusTilesObjs = selfPlayedTiles.filter(
+      (playedTiles) => playedTiles.connectionId !== mjPlayer.getConnectionId(),
+    );
+    const opponentsBonusTiles = opponentsBonusTilesObjs.map((playedTileObj: SelfPlayedTile) => {
+      const tilesArray = playedTileObj.playedTiles;
+      return tilesArray.map((strDef) => TileFactory.createTileFromStringDef(strDef));
+    });
+    mjOpponents.forEach((opponent, index) => opponent.addPlayedTiles(opponentsBonusTiles[index]));
```

Comment:
I see you're matching up the indices of the playedTiles with mjopponents. May be risky if states somehow changed. Maybe do 
opponentsBonusTiles.forEach((playedTileObj) =>{ find correct opponent in mjOpponent, and add played tiles }
Technically n^2, but if we care about efficiency (of n = 3 lol), we can always map it to {connectionId: MjOpponent }. Up to you.

Also, use opponent.addSelfPlayedTiles() addPlayedTiles is for creating melds from deadpile

Reply:
`3 opponents ^ 2 = 9` vs `3`, not too bad haha